### PR TITLE
Navbar and landing page added

### PR DESCRIPTION
**Pull request by Zeref101**
Navbar with native view, mobile view and tablet view added
landing page with responsiveness added
type->index.ts for type of links created
constants->index.ts for the navbar links created
custom tailwind classes created

### CODE REVIEW

**Review by WizzyGeek — APPROVED** [review]

File: src/components/LeftSideBar.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+				{leftSidebarLinks.map((link) => {
+					const isActive = pathname === link.route;
+					return (
+						<NavLink
+							key={link.label}
+							to={link.route}
+							className={`${
+								isActive
+									? "primary-gradient text-light-900 rounded-lg"
+									: "text-dark300_light900"
```
Comment (nexxeln) [inline src/components/LeftSideBar.tsx:26]: i think there's a mistake here, `text-dark300z_light900`?

**Review by nexxeln — CHANGES_REQUESTED** [review]
we have a lot of hardcoded hex values, would be better to extract them into css variables and then referencing them in our tailwind config so that we can use an abstracted class to use those colours.

**Comment by WizzyGeek** [comment]
@Zeref101 can you run `npm run format` and commit to pass the linting check?

**Comment by WizzyGeek** [comment]
Also correct me if I am wrong, but I don't think this is quite what @nexxeln described.

I am not quite fond of it too.

@nexxeln what's your opinion?

**Review by nexxeln — CHANGES_REQUESTED** [review]
okay yeah this im not a big fan of. what i meant was to put the colors we're using throughout the project in the `theme` object of the tailwind config. see these [docs on extending the tailwind config](https://tailwindcss.com/docs/theme#extending-the-default-theme

after you've done that revert back to what we had before, we don't need to over abstract our tailwind classes.

**Comment by WizzyGeek** [comment]
@Zeref101 please revert the last 3 commits

**Comment by WizzyGeek** [comment]
Just `git reset --hard 0a52fce && git push --force` will work good

File: tailwind.config.js [hunk 2]
```diff
@@ -1,8 +1,23 @@
 /** @type {import('tailwindcss').Config} */
-export default {
-	content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx}"],
+module.exports = {
+	content: [
+		'./pages/**/*.{ts,tsx}',
+		'./components/**/*.{ts,tsx}',
+		'./app/**/*.{ts,tsx}',
+		'./src/**/*.{ts,tsx}',
+	],
```
Comment (WizzyGeek) [inline tailwind.config.js:8]: Revert this to what we had earlier

**Review by WizzyGeek — APPROVED** [review]
Let's extract the colours into variables in another PR.

**Review by nexxeln — APPROVED** [review]
lgtm

**Comment by nexxeln** [comment]
interesting we're getting a build error

File: tailwind.config.js [hunk 3]
```diff
@@ -1,8 +1,18 @@
...
-export default {
+module.exports = {
 	content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx}"],
+
 	theme: {
-		extend: {},
+		colors: {
+			'sky-blue': '#78CBFF',
+			'midnight-blue': '#002133',
+		},
```
Comment (nexxeln) [inline tailwind.config.js:9]: this `colors` object should be in the `extend`

**Review by nexxeln — CHANGES_REQUESTED** [review]

**Review by nexxeln — APPROVED** [review]
lgtm, great work!

**Pull request merged**